eclipsesource/theia
# Stub vscode 'Notebook' API

#### What it does

_Stub vscode 'Notebook' API_
This enables the possibility to load and start plugins which depend on the 'Notebook' API.  The 'Notebook' functionality will not be operational but the rest of the functionality offered by the plugin will be.

Fixes #11786

Contributed on behalf of ST Microelectronics


#### How to test 
1. Install extension[ GitHub Issue Notebooks](https://open-vsx.org/extension/ms-vscode/vscode-github-issue-notebooks) from open-vsx, by downloading the vsix file and installing the downloaded file from extension view (note: tested on 0.0.127 on my side).  
2. Reproduce the problem in the theia instance before switching to this PR. Try to create a new Github Issues notebook (`ctrl+shift+p` and type `New Github Issue Notebook`)
4. An error should be displayed: `Activating extension 'GitHub Issue Notebooks' failed: vscode__WEBPACK_IMPORTED_MODULE_0__.workspace.onDidOpenNotebookDocument is not a function`
5. Update your `theia` repository to use this PR
6. Open the same workspace and make sure the message above is no longer displayed when creating a new Github issues notebook. Nothing should happen, as notebook API is only stubbed, not implemented.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
